sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverClient.java

```diff
@@ -41,8 +42,11 @@
         .setIsAutoCompleteEnabled(false)
         .setMaxAutoLockRenewalDuration(Duration.ZERO);
 
-    private final AtomicReference<EmitterProcessor<ServiceBusReceivedMessageContext>> messageProcessor =
+    private final AtomicReference<EmitterProcessor<ServiceBusReceivedMessageContext>> messageSource =
```

Comment:
Why is this an atomic reference? If I remember correctly you are synchronizing on a lock. Only one thread can be in the method.

Reply:
It is because checkstyle complain, I really do not need atomic reference because I synchronizing.

Reply:
What does checkstyle complain about?

Reply:
This is the error we get: The variable field 'messageSource ' of class 'implements' should be final. Classes annotated with @ServiceClient are supposed to be immutable. [ServiceClient]

And we can not define it as final, since we need to init it outside of constructor .